### PR TITLE
fix: Respect tab-complete aliases (Fixes #3266)

### LINKED ISSUE #3266: Tab Complete Alias - Bug
### Server Implementation

Paper

### Server Version

1.16.5

### Describe the bug

Aliases get tab completion (Disabled)

### To Reproduce

1. Install PlotSquared on 1.16.5
2. Remove aliases in settings.yml from "tab-completed-aliases"
3. Save file
4. Restart Server
5. Try to tab complete the removed aliases

### Expected behaviour

No tab completion possible

### Screenshots / Videos

image
image

### Error log (if applicable)

_No response_

### Plot Debugpaste

https://athion.net/ISPaster/paste/view/6a0dcde21ac448a3ba1b33b6914182e4

### PlotSquared Version

PlotSquared-6.1.2-Premium

### Checklist

- [X] I have included a Plot debugpaste.
- [X] I am using the newest build from https://www.spigotmc.org/resources/77506/ and the issue still persists.

### Anything else?
Discord (for questions): jo#1081

### PR DESCRIPTION
**Pull request by PierreSchwang**
## Overview
**Fixes https://github.com/IntellectualSites/PlotSquared/issues/3266**

## Description
PS now respects the `tab-completed-aliases` config entry properly

## Checklist

- [x] I included all information required in the sections above
- [x] I tested my changes and approved their functionality
- [x] I ensured my changes do not break other parts of the code
- [x] I read and followed the [contribution guidelines](https://github.com/IntellectualSites/PlotSquared/blob/v6/CONTRIBUTING.md)

### CODE REVIEW

**Review by NotMyFault — COMMENTED** [review]
To make it more clear for, is this (supposed to be) an opt-out list? Based on the description, you can add command prefixes that should be tab completed, yet atm you can only remove entries, new ones aren't recognized.

**Comment by PierreSchwang** [comment]
@NotMyFault There is already the config entry as described by `Command aliases that will be tab completed` - This didn't worked at all. Now they are checked, by default all aliases are in the config and therefor can be used for tab completion. if aliases are removed their sub commands won't be completed anymore

**Review by NotMyFault — APPROVED** [review]

File: Core/src/main/java/com/plotsquared/core/plot/flag/types/ListFlag.java [hunk 1]
```diff
@@ -43,8 +43,18 @@ public ListFlag(final List<V> valueList, final Caption category, final Caption d
...
+            if (!mergedList.contains(v)) {
+                mergedList.add(v);
+            }
+        }
+        // Only add new values if not already present from #getValue()
+        for (final V v : newValue) {
+            if (!mergedList.contains(v)) {
+                mergedList.add(v);
+            }
+        }
```
Comment (dordsor21) [inline Core/src/main/java/com/plotsquared/core/plot/flag/types/ListFlag.java:57]: Not a particular fan of this implementation. The order shouldn't need to be preserved (it isn't really preserved when merging anyway) so other collection implementations (Set) might be better for code conciseness.

Comment (dordsor21) [reply Core/src/main/java/com/plotsquared/core/plot/flag/types/ListFlag.java]: Updating fixed that

**Review by dordsor21 — APPROVED** [review]

**Pull request merged**